src/coreclr/src/utilcode/pedecoder.cpp

```diff
@@ -2335,6 +2335,41 @@ READYTORUN_HEADER * PEDecoder::FindReadyToRunHeader() const
     return NULL;
 }
 
+#ifndef DACCESS_COMPILE
+uint32_t PEDecoder::GetExport(LPCSTR exportName) const
+{
+    // Get the export directory entry
+    PIMAGE_DATA_DIRECTORY pExportDirectoryEntry = GetDirectoryEntry(IMAGE_DIRECTORY_ENTRY_EXPORT);
+    if (pExportDirectoryEntry->VirtualAddress == 0 || pExportDirectoryEntry->Size == 0)
+    {
+        return 0;
+    }
+    
+    const uint8_t *imageBase = (const uint8_t *)GetBase();
+    const IMAGE_EXPORT_DIRECTORY *pExportDir = (const IMAGE_EXPORT_DIRECTORY *)GetDirectoryData(pExportDirectoryEntry);
+    
+    uint32_t namePointerCount = VAL32(pExportDir->NumberOfNames);
+    uint32_t addressTableRVA = VAL32(pExportDir->AddressOfFunctions);
+    uint32_t namePointersRVA = VAL32(pExportDir->AddressOfNames);
+
+    for (uint32_t nameIndex = 0; nameIndex < namePointerCount; nameIndex++)
+    {
+        uint32_t namePointerRVA = VAL32(*(const uint32_t *)&imageBase[namePointersRVA + sizeof(uint32_t) * nameIndex]);
```

Comment:
I'm still concerned that these will be an out of bounds access for corrupted/maliciously-crafted binaries.

Reply:
I have simplified this code based on explicit Jan's request. Without being completely familiar or trying to interpret the official view, I think that the rule of thumb is that making sure binaries haven't been tampered with is not a matter for this low-level runtime, that's a question of deployment options, signatures etc. There are cases where even a legal PE file can crash the runtime, e.g. upon random disk sector failure. I defer to @jkotas for additional thoughts in this matter.